Sources/TuistKit/Services/DumpService.swift

```diff
@@ -17,6 +19,15 @@ final class DumpService {
         } else {
             projectPath = AbsolutePath.current
         }
+        let configLoader = ConfigLoader(
```

Comment:
@kwridan recently did some [work](https://github.com/tuist/tuist/pull/2692/files#diff-dd3bdd4af1c1c04d77a525722728758bef31bbac7b99135459a9e5de6e0aac7fR61) to encapsulate the plugin and graph loading in a component that we can use from all the services that need to load the graph. I'm wondering if we should do something similar here or reuse that utility that Kas implemented. Having similar logic in two places is a recipe for inconsistencies.

Reply:
I agree, I'd probably add it to that component

Reply:
Thanks for the review! 🙌 
If I use the helper you mentioned everything is much simpler, but I get a ValueGraph, which potentially contains multiple projects (`public var projects: [AbsolutePath: Project]`).
2 problems:
- the old implementation only had 1 project, should I serialize the dictionary of projects instead? This would break the output with respect to the old versions, is it a problem?
- the `Project` of the `ValueGraph` is not `Codable` , should I make it (and all the types it contains) `Codable`?

Reply:
You're right, using `ManifestGraphLoader` `loadGraph` function is not ideal. I'd make `loadPlugins` from that component public and in `DumpService` call `loadPlugins()` and then `try manifestLoader.loadProject(at: projectPath)`. This will ensure that `dump` works in the same way as it did before while reusing most of the logic.

We don't want to make `ValueGraph` `Codable`, at the moment at least.

Reply:
@pepibumur @fortmarek should be addressed now 🙌